Now the failure should appears:

```
00:21:26 ======================================================================
00:21:26 FAIL: test_hardshrink (test_torch.TestTorch)
00:21:26 ----------------------------------------------------------------------
00:21:26 Traceback (most recent call last):
00:21:26   File "/var/lib/jenkins/workspace/test/test_torch.py", line 5592, in test_hardshrink
00:21:26     self.assertEqual(torch.tensor([1, 0, 0, 0.6]).view(2, 2), data.hardshrink())
00:21:26   File "/var/lib/jenkins/workspace/test/common.py", line 307, in assertEqual
00:21:26     assertTensorsEqual(x, y)
00:21:26   File "/var/lib/jenkins/workspace/test/common.py", line 299, in assertTensorsEqual
00:21:26     self.assertLessEqual(max_err, prec, message)
00:21:26 AssertionError: tensor(0.5000) not less than or equal to 1e-05
00:21:26 
00:21:26 ----------------------------------------------------------------------
00:21:26 lambd = 0.3
00:21:26 lambd = 0.5
00:21:26 lambd = 0
00:21:26 Ran 508 tests in 29.749s
00:21:26 
00:21:26 FAILED (failures=1, skipped=123)
00:21:26 Traceback (most recent call last):
00:21:26   File "test/run_test.py", line 344, in <module>
00:21:26     main()
00:21:26   File "test/run_test.py", line 336, in main
00:21:26     raise RuntimeError(message)
00:21:26 RuntimeError: test_sparse failed!
```

--- a/aten/src/ATen/native/Activation.cpp
+++ b/aten/src/ATen/native/Activation.cpp
@@ -34,7 +34,6 @@ Tensor & rrelu_(Tensor & self, Scalar lower, Scalar upper, bool training, Genera
 }
 
 Tensor hardshrink_cpu(const Tensor & self, Scalar lambd) {
-  printf("lambd = %g\n", lambd.toTensor().toType(CPU(kDouble)).data<double>()[0]);
   auto lambd_tensor = lambd.toTensor().toType(self.type().scalarType()).toBackend(self.is_cuda() ? Backend::CUDA : Backend::CPU);
   auto out_tensor = at::empty_like(self);
   AT_DISPATCH_FLOATING_TYPES(self.type(), "hardshrink_cpu", [&] {

--- a/aten/src/ATen/native/native_functions.yaml
+++ b/aten/src/ATen/native/native_functions.yaml
@@ -845,12 +845,14 @@
 
 - func: relu_(Tensor self) -> Tensor
 
-- func: hardshrink(Tensor self, Scalar lambd=0.5) -> Tensor
+# TODO: there is a bug in float values of Scalar, after bug fixes, set the default value of lambd=0.5
+# for now, getting around this with default values defined in torch/nn/functional.py
+- func: hardshrink(Tensor self, Scalar lambd) -> Tensor
   dispatch:
     CPU: hardshrink_cpu
     CUDA: hardshrink_cuda
 
-- func: hardshrink_backward(Tensor grad_out, Tensor self, Scalar lambd=0.5) -> Tensor
+- func: hardshrink_backward(Tensor grad_out, Tensor self, Scalar lambd) -> Tensor
   dispatch:
     CPU: hardshrink_backward_cpu
     CUDA: hardshrink_backward_cuda

--- a/test/test_torch.py
+++ b/test/test_torch.py
@@ -5555,15 +5555,15 @@ class TestTorch(TestCase):
         ]
         for t in float_types:
             data = data_original.type(t)
-            self.assertEqual(torch.tensor([1, 0.5, 0, 0.6]).view(2, 2), data.hardshrink(0.3))
-
-            self.assertEqual(torch.tensor([1, 0, 0, 0.6]).view(2, 2), data.hardshrink(0.5))
-
-            # TODO: test default lambda (0.5)
-            self.assertEqual(torch.tensor([1, 0, 0, 0.6]).view(2, 2), data.hardshrink())
+            self.assertEqual(torch.tensor([1, 0.5, 0, 0.6]).view(2, 2), torch.nn.Hardshrink(0.3)(data))
+            self.assertEqual(torch.tensor([1, 0, 0, 0.6]).view(2, 2), torch.nn.Hardshrink(0.5)(data))
+            self.assertEqual(torch.tensor([1, 0, 0, 0.6]).view(2, 2), torch.nn.Hardshrink()(data))
 
             # test non-contiguous case
-            self.assertEqual(torch.tensor([1, 0.3, 0.5, 0.6]).view(2, 2), data.t().hardshrink(0.1))
+            self.assertEqual(torch.tensor([1, 0.3, 0.5, 0.6]).view(2, 2), torch.nn.Hardshrink(0.1)(data.t()))
+
+            # not supporting default lambd value for torch.hardshrink() due to a Scalar bug
+            self.assertRaises(TypeError, lambda: data.hardshrink())
 
     def test_unbiased(self):
         tensor = torch.randn(100)

